### ISSUE TITLE
Ability to lock certain pages for editing.

### ISSUE THREAD

**Issue by taamrove** [bitfocus/companion#2483]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

I think a function for locking certain pages would be nice for the applications where companion is an integrated part of a system.
I am now sitting on the equipment of one of the AV sister companies for BitFocus, and they have several racks that are setup with a "plug and play" setup, with "reset to default" settings and everything, but I think it would make sense when you have a intricate page and button setup, to lock the core of the pages, and rather have open pages that users can customise and reference the other pages/buttons and/or make their own from scratch.

So bottom line, ability to either lock buttons or pages for editing.
